src/main/java/io/debezium/connector/vitess/pipeline/txmetadata/ShardLineage.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector.vitess.pipeline.txmetadata;
+
+import java.util.Map;
+
+/**
+ * Class used to determine which parents a shard range descended from. Used to set the epoch to the succeediing
+ * epoch of its parents.
+ */
+public class ShardLineage {
+
+    /**
+     * Return the epoch value of the shard, based on its parents epochs.
+     * If there are parents present, return the max of the parent epochs plus one.
+     * If there are no parents present, it returns zero.
+     *
+     * @param shardString The descendant shard to find parents of
+     * @param shardEpochMap The map to search for parents
+     * @return The epoch value of the descendant shard
+     */
+    public static Long getInheritedEpoch(String shardString, ShardEpochMap shardEpochMap) {
+        Shard shard = new Shard(shardString);
+
+        Long maxParentEpoch = -1L;
+        for (Map.Entry<String, Long> shardEpoch : shardEpochMap.getMap().entrySet()) {
+            String currentShardString = shardEpoch.getKey();
+            Long currentEpoch = shardEpoch.getValue();
+            Shard currentShard = new Shard(currentShardString);
+            if (shard.overlaps(currentShard)) {
+                maxParentEpoch = Math.max(maxParentEpoch, currentEpoch);
+            }
+        }
+
+        return maxParentEpoch + 1;
+    }
+
+    private static class Shard {
+
+        // A string lexicographically less than all other strings
+        public static final String NEGATIVE_INFINITY = "";
+        // A string lexicographically greater than all other strings
+        public static final String POSITIVE_INFINITY = "\uFFFF";
```

Comment:
Is 4F long/big enough, do we need 8F?

Reply:
Ah this `\uFFFF` is actually a single special char that is greater than all other chars. I used the Character package to make this clear (it is the MAX_VALUE). And it turns out there is MIN_VALUE which is another special char (more clear than empty string). I updated the comments to describe this too.